extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -105,9 +104,10 @@ private static Properties getStreamsProperties(Properties properties, String boo
         streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, runtimeConfig.applicationId);
 
         // app id
-        if (runtimeConfig.applicationServer.isPresent()) {
-            streamsProperties.put(StreamsConfig.APPLICATION_SERVER_CONFIG, runtimeConfig.applicationServer.get());
-        }
+        streamsProperties.put(StreamsConfig.APPLICATION_SERVER_CONFIG, runtimeConfig.applicationServer.isPresent()
+                ? runtimeConfig.applicationServer.get()
+                : String.format("%s:%s", getQualifiedHostName(),
+                        ConfigProvider.getConfig().getOptionalValue("quarkus.http.port", String.class).orElse("8080")));
```

Comment:
I guess we can safely ignore test/dev ports (since it would be a single instance in these cases), but should we also consider quarkus.http.sslPort in case SSL is enabled (and insecure requests may even be disabled) ? But it starts to be a bit complex (cf options in https://github.com/quarkusio/quarkus/blob/master/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java) ^^